frontend/src/metabase/databases/components/DatabaseDetailField/DatabaseDetailField.tsx

```diff
@@ -89,9 +115,17 @@ const getPasswordProps = (field: EngineField) => {
 
 const getSelectProps = (field: EngineField, override?: EngineFieldOverride) => {
   return {
-    options: override?.options ?? field.options ?? [],
+    data: convertEnginFeildOptionsToSelectOptions(
+      override?.options ?? field.options ?? [],
+    ),
   };
 };
 
+const convertEnginFeildOptionsToSelectOptions = (
```

Comment:
nit - spelling: `convertEngineFieldOptionsToSelectOptions`

since this happens in two places involving EngineOptions, would it make sense to add to `engine.ts`?

(sorry if i'm overstepping as a random outsider)